integration-tests/suites/common/executor.go

```diff
@@ -182,6 +184,22 @@ func (e *executor) RunCommand(cmd *exec.Cmd) (string, error) {
 	return trimmed, err
 }
 
+func (e *executor) ExecWithStdin(pippedContent string, args ...string) (res string, err error) {
+	cmd := e.builder.ExecCommand(args...)
+
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return "", err
+	}
+
+	go func() {
+		defer stdin.Close()
```

Comment:
How does ownership work in this case? Is `stdin` copied into this goroutine and is therefore safe if the "outer" function returns?

Reply:
My understanding is that the anonymous function is a usual closure, and shares stdin with the outer scope. Things are even simpler here since both the lambda and the defer call happen before the outer function returns.